packages/aws-cdk-lib/aws-codebuild/lib/project.ts

```diff
@@ -1361,7 +1364,7 @@ export class Project extends ProjectBase {
     return {
       type: this.buildImage.type,
       image: this.buildImage.imageId,
-      imagePullCredentialsType: imagePullPrincipalType,
+      imagePullCredentialsType: this.isLambdaBuildImage(this.buildImage) ? undefined : imagePullPrincipalType,
```

Comment:
```suggestion
      imagePullCredentialsType: imagePullPrincipalType,
```
Will remain unchanged after validating the image.